### PR TITLE
PIN-1856: add notification projection

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogEventsSerde.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    case `catalogItemDocumentUpdated`        => _.convertTo[CatalogItemDocumentUpdated]
+    case `catalogItemDeleted`                => _.convertTo[CatalogItemDeleted]
+    case `catalogItemDocumentAdded`          => _.convertTo[CatalogItemDocumentAdded]
+    case `catalogItemDocumentDeleted`        => _.convertTo[CatalogItemDocumentDeleted]
+    case `catalogItemDescriptorAdded`        => _.convertTo[CatalogItemDescriptorAdded]
+    case `catalogItemDescriptorUpdated`      => _.convertTo[CatalogItemDescriptorUpdated]
+  }
+
+  def getKind(e: Event): String = e match {
+    case CatalogItemAdded(_)                     => catalogItemAdded
```
Comment (galales) [inline models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogEventsSerde.scala]:
```suggestion
    case _: CatalogItemAdded                     => catalogItemAdded
```
unapply not needed here

File: src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/projection/CatalogNotificationProjection.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package it.pagopa.interop.catalogmanagement.model.persistence.projection
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/projection/CatalogNotificationProjection.scala:1]: may have sense to move this to commons?

**Review by galales — APPROVED** [review]

File: models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogEventsSerde.scala (hunk 1 above)
Comment (beetlecrunch) [reply models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogEventsSerde.scala]: LGTM, but I suggest applying the same pattern in all the mgmt (e.g. purpose_manangement use unapply)

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/projection/CatalogNotificationProjection.scala (hunk 2 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/projection/CatalogNotificationProjection.scala:1]: +1

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/projection/CatalogNotificationProjection.scala:1]: Looks good to me, but in a new PR.

**Review by TonioGela — APPROVED** [review]

**Pull request merged**